tcs/fn_initPlayer.sqf

```diff
@@ -19,10 +19,33 @@ if (TCS_var_addSurgicalKitsToMedics) then {
 	};
 };
 
-if (TCS_var_saveAndRestoreLoadouts) then {
-	TCS_var_playerLoadout = getUnitLoadout player;
+// Wait for ACRE to initialize any editor-given radios.
+waitUntil {
+	sleep 0.5;
+	[] call acre_api_fnc_isInitialized;
 };
 
+// Remove the player's editor-given radios, to avoid ID collision on respawn.
+_radioList = [] call acre_api_fnc_getCurrentRadioList;
+[_radioList] call TCS_fnc_removePlayerRadios;
+
+// Save the voiceless, radioless loadout to TCS variable
+TCS_var_playerLoadout = getUnitLoadout player;
+
+// Save the list of editor-given radios with IDs removed to TCS variable
+TCS_var_playerRadios = [];
+{
+	_regout = _x regexFind ["(.*?)_ID"];
+	_radio = _regout select 0 select 1 select 0;
+	TCS_var_playerRadios append [_radio];
+} forEach _radioList;
+
+// Re-add any radios that were given in the editor.
+{
+	(uniformContainer player) addItemCargoGlobal [_x, 1];
+} forEach TCS_var_playerRadios;
+
+// Give framework radios

```

Comment:
Couldn't this be in `tcs_fnc_initPlayerRadios` since this would be part of it? Because ideally this file, for readability and making it easier to understand, could be just a "checklist" of what's done for every player and having a lot of code here makes it harder at a glance to figure out what is done when.

There is also a simpler way of getting a radio class without an ID: https://github.com/TheCoolerServer/tcs-framework/blob/master/tcs/radios/fn_savePlayerRadioSettings.sqf#L15

And just to be sure, what we are doing is:
1. Getting the radios the player has and saving them
2. Removing those radios 
3. Saving the loadout without those radios
4. Adding them back in
5. Applying the default radios?

Is there a reason we need to do all this?

Reply:
Can't we test that? If the loadouts are getting saved with ids we could load the editor, wait for ACRE to initialize and call `getUnitLoadout` and see if the radios are being saved with IDs there or not.